### PR TITLE
mraa's GPIO to work on SLES for Raspberry Pi

### PR DESCRIPTION
**Pull request by cbosdo**
There are a number of things that are different between raspbian and other distros on Raspberry Pi. Overcame these to get mraa's gpio working on SLES (and openSUSE)

### CODE REVIEW

File: src/arm/arm.c [hunk 1]
```diff
@@ -87,6 +87,8 @@ mraa_arm_platform()
             platform_type = MRAA_96BOARDS;
         else if (mraa_file_contains("/proc/device-tree/model", "s900"))
             platform_type = MRAA_96BOARDS;
+        else if (mraa_file_contains("/sys/firmware/devicetree/base/compatible", "raspberrypi,"))
```
Comment (alext-mkrs) [inline src/arm/arm.c]: `/sys/firmware/devicetree/base` should be `/proc/device-tree` to align with the kernel recommendation and the very change one of the other patches in this PR introduces.

**Comment by cbosdo** [comment]
@alext-mkrs nice catch! I forgot about this one, and updated my branch with the proper value.

**Comment by arfoll** [comment]
Merged! This is awesome, thanks alot for this.

Btw do you have a mraa project on suse's OBS? Myself and @tingleby where discussing doing this but if you have something already we could just link to it in our front page where we have our PPA link.

**Comment by cbosdo** [comment]
> Btw do you have a mraa project on suse's OBS?

We have one with the patches from this PR in the [hardware project](https://build.opensuse.org/package/show/hardware/mraa)

**Comment by arfoll** [comment]
That's awesome, are you happy with us linking to these repos from the top
level README.md file?

I'll make a mraa release soon so you don't have to keep those patches like
this :)

**Comment by cbosdo** [comment]
yeah sure, that'ld be awesome!

**Comment by cbosdo** [comment]
@arfoll done, see PR #657

**Comment by g-vidal** [comment]
Hello, 
I have been using mraa on a raspberrypi 3 with a debian stretch without anay problem for months, but I have not been able to get an mraa functional image with a kernel younger than 4.4.37-v7+ . Issue #620 was closed without solution. I hoped  these contributions may have helped but  no change. Could you have please some advice or hint on how to compile an image with mraa and  a more recent kernel  on araspberrypi ? I do not want to go back to the dark corners of raspbian and things are working very well with the old kernel  [raspi3image](http://mediaserv.climatetmeteo.fr/images/RaspBerry/DebianStretchPi3/) I have merged swig latest version with @arfoll  amandments and the compilation works  with Node 7.7.

Thanks for any help

**Pull request closed without merging**